Source/engine/render/text_render.cpp

```diff
@@ -92,14 +92,38 @@ text_color GetColorFromFlags(UiFlags flags)
 	return ColorWhitegold;
 }
 
-} // namespace
+std::array<uint8_t, 256> *LoadFontKerning(GameFontTables size, uint16_t row)
+{
+	uint32_t fontId = (size << 16) | row;
+
+	auto hotKerning = FontKerns.find(fontId);
+	if (hotKerning != FontKerns.end()) {
+		return hotKerning->second.get();
+	}
+
+	char path[32];
+	sprintf(path, "fonts\\%i-%02d.bin", FontSizes[size], row);
+
+	auto kerning = std::make_unique<std::array<uint8_t, 256>>();
+
+	LoadFileInMem(path, kerning.get());
+
+	return FontKerns.insert(make_pair(fontId, move(kerning))).first->second.get();
+}
 
-void LoadFont(GameFontTables size, text_color color)
+Art *LoadFont(GameFontTables size, text_color color, uint16_t row)
 {
-	auto font = std::make_unique<Art>();
+	uint32_t fontId = (color << 24) | (size << 16) | row;
+
+	auto hotFont = Fonts.find(fontId);
+	if (hotFont != Fonts.end()) {
+		return hotFont->second.get();
+	}
 
 	char path[32];
-	sprintf(path, "fonts\\%i-00.pcx", FontSizes[size]);
+	sprintf(path, "fonts\\%i-%02d.pcx", FontSizes[size], row);

```

Comment:
`LoadFileInMem()` triggers a fatal error if I type a character that's not supported.

![image](https://user-images.githubusercontent.com/9203145/134284030-3dd072ec-e4ef-43d3-88ce-b7c0b9835b29.png)

Reply:
yeah, plan is to add the rest, but maybe we should simply skip any such missing symbols :thinking:

Reply:
@AJenbo I would vote for writing a error character. A questions mark or a box or something.
The program shouldn't stop executing or simply print nothing.
That could also help to diagnostic future errors.

Reply:
I guess that might be safe to do considering that UTF-8 is self synchronizing. But I would go for `?` as that is more canonical and simpler to add to the code.